src/cassandra/statement.rs

```diff
@@ -727,41 +739,45 @@ impl Statement {
         }
     }
 
-    //  ///Bind a "decimal" to a query or bound statement at the specified index.
-    //    pub fn bind_decimal(&self,
-    //                                index: i32,
-    //                                value: d128)
-    //                                -> Result<&mut Self, CassError> {
-    //            unsafe {
-    //                CassError::build(
-    //                    cass_statement_bind_decimal(
-    //                        self.inner(),
-    //                        index,
-    //                        value
-    //                    )
-    //                ).wrap(&mut self)
-    //            }
-    //        }
-
-    // Binds a "decimal" to all the values with the specified name.
-    //
-    // This can only be used with statements created by
-    // cass_prepared_bind().
-    //    pub fn bind_decimal_by_name<'a>(&'a self,
-    //                                    name: &str,
-    //                                    value: String)
-    //                                    -> Result<&'a Self, CassError> {
-    //        unsafe {
-    //            let name = CString::new(name).unwrap();
-    //            CassError::build(
-    //            cass_statement_bind_decimal_by_name(
-    //                self.inner(),
-    //                name.as_ptr(),
-    //                value
-    //            )
-    //        ).wrap(&self)
-    //        }
-    //    }
+    /// Binds a "BigDecimal" to a query or bound statement at the specified index.
+    pub fn bind_decimal(&mut self, index: usize, value: &BigDecimal) -> Result<&mut Self> {
+        let dec_parts = value.as_bigint_and_exponent();
+        let varint = dec_parts.0.to_signed_bytes_be();
+        let scale = dec_parts.1;
+
+        //@TODO: cpp scale = i32, BigInt scale = i64
```

Comment:
Thanks! I think it's highly unlikely anyone will be using 4GB BigInts, but just in case, you should convert scale using try_into() and then if it fails return an error. I think it's OK to reuse one of the `CASS_ERROR_LIB` errors here, even though that's not strictly accurate: probably `CASS_ERROR_LIB_INVALID_DATA`.